src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/endpoints/JurisdictionEndpoint.java

```diff
@@ -0,0 +1,84 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.endpoints;
+
+import feign.FeignException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.actuate.endpoint.annotation.Endpoint;
+import org.springframework.boot.actuate.endpoint.annotation.ReadOperation;
+import org.springframework.boot.actuate.endpoint.annotation.Selector;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.idam.Credential;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.idam.JurisdictionToUserMapping;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.idam.NoUserConfiguredException;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
+
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.util.Map.Entry;
+
+@Component
+@Endpoint(id = "jurisdictions")
+@EnableConfigurationProperties(JurisdictionToUserMapping.class)
+public class JurisdictionEndpoint {
+
+    private static final Logger log = LoggerFactory.getLogger(JurisdictionEndpoint.class);
+
+    private final Map<String, Credential> jurisdictions;
+    private final Function<String, Credential> credential;
+    private final IdamClient idamClient;
+
+    public JurisdictionEndpoint(
+        JurisdictionToUserMapping mapping,
+        IdamClient idamClient
+    ) {
+        jurisdictions = mapping.getUsers();
+        credential = mapping::getUser;
+        this.idamClient = idamClient;
+    }
+
+    @ReadOperation
+    public Map<String, HttpStatus> jurisdictions() {
+        return jurisdictions
+            .entrySet()
+            .stream()
+            .collect(Collectors.toMap(
+                Entry::getKey,
+                entry -> checkCredentials(entry.getValue())
+            ));
+    }
+
+    @ReadOperation
+    public HttpStatus jurisdiction(@Selector String jurisdiction) {
+        try {
+            return checkCredentials(credential.apply(jurisdiction));
+        } catch (NoUserConfiguredException exception) {
+            return HttpStatus.UNAUTHORIZED;
+        }
+    }
+
+    private HttpStatus checkCredentials(Credential credential) {
+        try {
+            idamClient.authenticateUser(credential.getUsername(), credential.getPassword());
+
+            log.debug("Successful authentication");
+        } catch (FeignException exception) {
+            logException(credential.getUsername(), exception);
+
+            return HttpStatus.valueOf(exception.status());
+        } catch (Exception exception) {
+            logException(credential.getUsername(), exception);
+
+            return HttpStatus.UNAUTHORIZED;
```

Comment:
more like 500 imho

Reply:
I've reverted these changes as following reviewer had different concerns. And the size of concerns already raises the suggestion we should discuss in person what team desires to respond in each situation